### ISSUE TITLE
Label leakage: use mutual information

### ISSUE THREAD

**Issue by angela97lin** [alteryx/evalml#927]
Per @kmax12's comment in #917, in `DetectLabelLeakage` data check we could use mutual information which can support numeric and categorical data instead of pearson correlation.

https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_classif.html
https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_regression.html#sklearn.feature_selection.mutual_info_regression

**Comment by angela97lin** [comment]
Not currently working on this so putting this back in the backlog for now!

**Comment by kmax12** [comment]
fyi, i think woodwork will have functionality for calculating mutual info we can potentially leverage for this

**Comment by angela97lin** [comment]
Yup, [I see it in the latest release](https://woodwork.alteryx.com/en/latest/generated/woodwork.data_table.DataTable.get_mutual_information.html#woodwork.data_table.DataTable.get_mutual_information). Thanks for the heads-up! 😊
